context/src/main/java/io/micronaut/scheduling/processor/ScheduledMethodProcessor.java

```diff
@@ -93,6 +94,10 @@ public void process(BeanDefinition<?> beanDefinition, ExecutableMethod<?, ?> met
         if (!(beanContext instanceof ApplicationContext)) {
             return;
         }
+        Class<Object> beanType = (Class<Object>) beanDefinition.getBeanType();
+        if (!beanContext.containsBean(beanType, ExactTypeQualifier.INSTANCE)) {
```

Comment:
this should be `beanDefinition.getDeclaredQualifier()` and probably `beanDefinition.asArgument()` for the type

Reply:
I tried this approach and it does not work unfortunately. For the beanDefinition of the original bean `beanContext.getBean` returns the intercepted bean and `beanContext.containsBean()` is true. So both beans end up being scheduled.

What issue do you see with the current version?